sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/FeatureFlagConfigurationSetting.cs

```diff
@@ -152,51 +171,186 @@ public IList<FeatureFlagFilter> ClientFilters
         internal override void SetValue(string value)
         {
             _originalValue = value;
-
             _isValidValue = TryParseValue();
         }
 
         internal override string GetValue()
         {
-            return _isValidValue ? FormatValue() : _originalValue;
+            // If the setting was created using the composite constructor, it
```

Comment:
FWIW, it feels a little brittle to handle the data via multiple code paths and need to switch on different logic depending on which constructor was called.  If there's a strong reason to need to use this approach, that's fine, but I'm wondering if there's a way to simplify that would make the code easier to support and maintain in the future.

Reply:
There probably is, but this pattern was in place.  An overhaul is out of scope for the change that I'm intending to make.  If we'd like to do so, I'll ask that we open an issue and treat it as a separate effort.

Reply:
Sure, I wasn't suggesting you change the top-level pattern.  I just think you might be able to address the issue without changing the implementation of this method.

Reply:
Changing the approach for parsing/writing took care of this naturally.   We know have two known states:

- The raw value was provided and does not conform to the schema, so we're treating it as a plain string.
- The setting uses structured properties that need to be written.